pkg/blob/controllerserver.go

```diff
@@ -310,6 +316,10 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		storageEndpointSuffix = d.getStorageEndPointSuffix()
 	}
 
+	if storeAccountKey && !pointer.BoolDeref(allowSharedKeyAccess, false) {
```

Comment:
I think here should be `!pointer.BoolDeref(allowSharedKeyAccess, true)` since if we didn't give allowSharedKeyAccess the default is true

Reply:
yes, the default value is true

Reply:
Technically allowSharedKeyAccess is never nil here but let me refactor the code a little to make it more straightforward.